### DISCUSSION TITLE
Params in query inside `getServerSideProps`

### DISCUSSION THREAD

**Discussion by akellbl4** [vercel/next.js#15863] [Help] [answered]
I'm trying to understand why routing params pass to `query` object in the context of  `getServerSideProps`.

For example:
```
/city/[city] -> /city/new-york?token=mytoken
```
Parsed as:
```
{
  query: { token: 'mytoken', city: 'new-york’ },
  params: { city: 'new-york' }
}
```

Is it a default behavior or maybe I did something wrong?
And if it was expected why it was made this way

**Comment by jamesmosier** [comment, marked as answer]
Yes that is the correct behavior from testing. 

I can't speak to the *why* because I did not make it, but looking at the [Dynamic Routing](https://nextjs.org/docs/routing/dynamic-routes) docs, the `query` object contains both dynamic route information as well as query string information. So this paradigm is likely just carried over to `getServerSideProps`.

**Reply by akellbl4** [reply]
This behavior is really strange for me because I'm trying to compare this behavior with any nodejs router for example `express` or `koa-router`.

Thanks for your answer.
